### ISSUE TITLE
Refactor: Replace "shells" to "repls"

### ISSUE THREAD

**Issue by anki-code** [xonsh/xonsh#5750]
When we're talking about xonsh shell types (the components that connect terminal and xonsh core) we're using term 'xonsh shell' or 'shell type' e.g.  prompt-toolkit shell, readline shell, run `xonsh --shell-type ptk`. 

There are so much confusion around this:
* We have terms 'xonsh shell', 'readline shell', 'ptk shell', 'shell', 'shell type' and this all make a mess: 'what is really shell?'.
* We have [`xonsh/shells/`](https://github.com/xonsh/xonsh/tree/main/xonsh/shells) subdirectory and this is not so good naming because it contains the code that connects terminal with prompt engine and with xonsh executor. It's not exactly 'the shell'.
* We have `__xonsh__.shell.shell.prompter.prompt` (shell-shell, prompt-prompt) and it would be more understandable to have something like `__xonsh__.shell.frontend.prompt`.
* The shell word is pretty high level and produces different expectations. I prefer to use 'shell' word only for call xonsh core i.e. xonsh shell means xonsh as a whole OR xonsh core (parser and executor).
* There are terms 'foreign shell' (e.g. `source-foreign` builtin), 'login shell' where we have 'shell' word again and again.
 
So we need to replace 'shell type', 'ptk shell', 'readline shell', 'xonsh/shell' to more clean name for this component.

What we want to achieve by renaming:
* More clear understanding and documentation.
* More clear addressing the idea that xonsh frontend can be adjusted, extended or rebuild.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by bestlem** [comment]
Isn't the common terminology for this that the shell is interactive.

From the bash quote below -  whose input and error output are both connected to terminals which I think matches xonsh's use of readline and ptk

From [zsh guide](https://zsh.sourceforge.io/Guide/zshguide02.html)
> In the second case, when you are typing at a prompt and waiting for each command to run, the shell is interactive;

[Bash](https://www.gnu.org/software/bash/manual/html_node/What-is-an-Interactive-Shell_003f.html) - slightly more complex 

> An interactive shell is one started without non-option arguments (unless -s is specified) and without specifying the -c option, whose input and error output are both connected to terminals (as determined by isatty(3)), or one started with the -i option.

https://unix.stackexchange.com/a/43389/5140

**Comment by gforsyth** [comment]
I agree that the word "shell" is overloaded and it can be confusing.  I think "frontend" is probably more appropriate in a vacuum, but I think that word is also overloaded and might cause confusion.

The "correct" term is probably "line editor", but that doesn't sound great and I think is _also_ confusing.

Of all of the terms, `prompt` is probably the best compromise (or `prompt-type`).

I'm not opposed to these changes, but I think we should keep the old flags available as aliases (or we can keep them enabled but unadvertised, so they wouldn't show up in `xonsh --help` but they would _work_ if used)

**Comment by anki-code** [comment]
### Use "repl" word instead of "shell" word

I've found a good well known word - REPL. PTK, Readline are REPLs e.g. for PTK:
* Read - all about how PTK setup and integrate into terminal.
* Eval - how PTK invoke execution and setup terminal before-after.
* Print - how PTK show the result: colors, above prompt, below prompt, etc.
* Loop - how PTK iterates input.
And this wording is pretty close with Python REPL.

### Xonsh Repls

So we need to rename Shell to Repl in xonsh code. After this we have Xonsh Repls: ptk, readline. We can run `xonsh --repl readline` and instead of `$SHELL_TYPE` we have `$XONSH_REPL="prompt-toolkit"`. It's very clear.

### Refactor `__xonsh__.shell.shell.prompter.prompt`

Next, we have `__xonsh__.shell.shell.prompter.prompt` - super confusing wording.

How it works:

* prompt_toolkit:
    - **`__xonsh__`** → `XonshSession` singleton (`xonsh/built_ins.py`)
    - **`.shell`** → `Shell` — the REPL dispatcher that picks a backend (`xonsh/shell.py`)
    - **`.shell`** → `PromptToolkitShell` — the actual PTK backend, subclass of `BaseShell` (`xonsh/shells/ptk_shell/__init__.py`)
    - **`.prompter`** → `prompt_toolkit.shortcuts.PromptSession` — holds key bindings, default buffer, history, clipboard
    - **`.prompt`** → bound method `PromptSession.prompt(**kwargs)` — runs the event loop, draws the prompt, returns the typed line

* readline:
    - **`__xonsh__`** → `XonshSession` singleton (`xonsh/built_ins.py`)
    - **`.shell`** → `Shell` — same dispatcher (`xonsh/shell.py`)
    - **`.shell`** → `ReadlineShell` — subclass of `BaseShell` + `cmd.Cmd` (`xonsh/shells/readline_shell.py`)
    - **`.prompter`** → *does not exist* — readline is a stdlib C module, configured globally via `readline.set_completer(...)`, `readline.parse_and_bind(...)` in `setup_readline()`
    
So
* `__xonsh__.shell` - is just manager that choose the repl on startup. We don't use it for any work. Ignore.
* `__xonsh__.shell.shell` - it's real repl (ptk or readline) - we need to have it and we put it to `__xonsh__.repl` or may be the more explicit way is to have `__xonsh__.repl.ptk`.
* `__xonsh__.shell.shell.***` is the certain repl internals.

### Result

As result we will have clear structure:

* xonsh
  * repl - ptk, readline, etc.
    * prompt - what is related to prompt line `user@host ~ @`
    * completer - all about completers.

No any intersections between "shell", "prompt" words.


### Future

In the future we will have PyREPL and it will be part of xonsh - https://github.com/xonsh/xonsh/issues/6255
